ykcompile/src/lib.rs

```diff
@@ -122,6 +121,12 @@ impl Location {
     }
 }
 
+enum RegisterVal {
```

Comment:
Total bikeshed, but I think the name of this could be better. `Val` suggests it expresses a value, but it actually expresses a register allocation decision.

How about `RegAlloc`?

Reply:
I tend to agree that `RegisterVal` is confusing (I misread it at first glance), not that I fully understand all of this anyway!

Reply:
No problem. I have no attachment to that name, `RegAlloc` works for me.

Reply:
Happy for me to fix this after squashing things down, since this will introduce conflicts in each commit.